include/envoy/network/dns.h

```diff
@@ -39,7 +39,8 @@ class DnsResolver {
    * @return if non-null, a handle that can be used to cancel the resolution.
    *         This is only valid until the invocation of callback or ~DnsResolver().
    */
-  virtual ActiveDnsQuery* resolve(const std::string& dns_name, ResolveCb callback) PURE;
+  virtual ActiveDnsQuery* resolve(const std::string& dns_name, uint32_t port,
```

Comment:
I agree. A new interface for SRV will be needed anyway if we are going to utilize the priority and weight info from SRV query results.